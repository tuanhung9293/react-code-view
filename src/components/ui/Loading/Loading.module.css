@value colors: 'styles/colors.css';
@value primary, black from colors;

.spinner{
  position: absolute;
  left: 50%;
  margin-left: -12px;

  width: 25px;
  height: 25px;
  border-radius: 50%;
  border-left: 3px solid primary;
  border-top: 3px solid primary;
  border-right: 3px solid black;
  border-bottom: 3px solid black;
  animation: loading-spinner 1s linear infinite;
}

@keyframes loading-spinner{
  from {
    transform: rotateZ(0deg);
  }
  to {
    transform: rotateZ(360deg);
  }
}