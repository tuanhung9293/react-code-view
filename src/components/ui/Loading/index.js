import React from 'react'
import styles from './Loading.module.css'

function Loading() {
  return <div className={styles.spinner} />
}

export default Loading