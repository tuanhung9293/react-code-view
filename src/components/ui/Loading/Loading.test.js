import React from 'react'
import {shallow} from 'enzyme'
import Loading from './'

it('renders Loading', () => {
  expect(shallow(
    <Loading />
  )).toMatchSnapshot()
})
