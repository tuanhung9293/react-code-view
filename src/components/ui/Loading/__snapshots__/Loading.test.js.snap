// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loading 1`] = `
<div
  className="spinner"
/>
`;
